src/main/scala/viper/silver/plugin/standard/termination/TerminationPlugin.scala

```diff
@@ -111,6 +111,34 @@ class TerminationPlugin(@unused reporter: viper.silver.reporter.Reporter,
    * Remove decreases clauses from the AST and add them as information to the AST nodes
    */
   override def beforeVerify(input: Program): Program = {
+    // Prevent potentially unsafe (mutually) recursive function calls in function postcondtions
+    // for all functions that don't have a decreases clause
+    val cycles = Functions.findFunctionCyclesVia(input, func => func.body.toSeq, func => func.body.toSeq)
```

Comment:
you could avoid computing cycles when `hasDecreasesClause` is true by moving this stmt into the then branch

Reply:
I kept it outside to avoid recomputing it for each function. But I made it a lazy val now, so hopefully it'll only be computed when needed.